lib/model/model.go

```diff
@@ -1338,6 +1339,15 @@ func (m *model) ccHandleFolders(folders []protocol.Folder, deviceCfg config.Devi
 	}
 
 	indexSenders.removeAllExcept(seenFolders)
+	// All current pending folders were touched above, so discard any with older timestamps
+	if expired, err := m.db.RemovePendingFoldersBeforeTime(deviceID, handleTime); err != nil {
+		l.Infof("Could not clean up pending folder entries: %v", err)
+	} else if expired > 0 {
+		m.evLogger.Log(events.ClusterPendingChanged, map[string]interface{}{
```

Comment:
I think this should also be emitted in places where we currently emit folder_rejected or device_rejected, to effectively have one event that represents all pending stuff changes.

Reply:
I am actually happy with additions / updates being handled without an extra API query and removals (or anything else) having a separate refresh trigger event. For consistency we should send the latter as well for additions. But I couldn't see a nice way for the consumer to avoid a needless refresh after already handling a FolderRejected event right before.

So shall we:

1. Send both events for a while, with one marked deprecated and having the same contents.

2. Switch the GUI to the new event immediately.

3. Handle contained detailed changes in-place, and fall back to an API refresh if they are not provided.

4. Remove the old event completely at some point.

The desired granularity is not obvious to me. We could for example unify FolderRejected and DeviceRejected, or keep them separate in the new structure?

Reply:
Thanks for your comments above @AudriusButkevicius, but what's your take on granularity? Coalesce addition and removal into one "changed" event I guess? But keep device and folder changes in separate events?

Reply:
I think it's fine to have it in a single event, as both deletions amd additions happen at the same time, upon connecting, so it's likely we'll have both additions and removals populated.

Reply:
Currently we send the `...Rejected` events for each thing individually.  You're suggesting to collect that and send one event per ClusterConfig?

Reply:
I suggest to leave the old event as it was, and just send the new one in the new format. You can even send multiple new ones, but they should allow the state to be event sourced.

Reply:
Hope the current draft meets that expectation.  Still needs some cleanup, but it works in principle, using the two new events (`Pending...Changed`).  Old ones (`...Rejected`) are in fact untouched, just no longer used in our GUI.